### PR TITLE
chore(storage): cleanup

### PR DESCRIPTION
**Pull request by ashwinkumar6**
#### Description of changes:
Update Storage list, copy, remove API docs

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/storage/js/copy.mdx [hunk 1]
```diff
@@ -7,19 +7,16 @@ success.
...
 
 const copyFile = async () => {
 	try {
 		const response = copy({
-			source: {
-				key: 'srcKey',
-				accessLevel: srcLevel
-			},
-			destination: { key: 'destKey, accessLevel: destLevel },
+			source: { key: 'srcKey' },
```
Comment (ashwinkumar6) [inline src/fragments/lib-v1/storage/js/copy.mdx:18]: I think we can provide a minimalistic example first

File: src/fragments/lib-v1/storage/js/copy.mdx [hunk 2]
```diff
@@ -34,19 +31,19 @@ You can copy a file from the specified key to another key within the same [File
 ```javascript
 import { copy, list } from 'aws-amplify/storage';
 
-console.log(await list({ path : 'copied/' })); // []
+console.log(await list({ prefix : 'copied/' })); // []
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/copy.mdx:34]: Hmmm, I think found this in previous PR but was not addressed🤔. It seems to lost during closing in-progress PR and creating new one. Let's try stick to 1 PR next time ☺️. cc @kvramyasri7

**Review by AllanZhengYP — APPROVED** [review]
Good catches! @ashwinkumar6

File: src/fragments/lib-v1/storage/js/copy.mdx (hunk 1 above)
Comment (kvramyasri7) [reply src/fragments/lib-v1/storage/js/copy.mdx:18]: I think you can keep the existing example but make `accessLevel` to be optional.

File: src/fragments/lib-v1/storage/js/copy.mdx (hunk 2 above)
Comment (kvramyasri7) [inline src/fragments/lib-v1/storage/js/copy.mdx:34]: Good catch.

File: src/fragments/lib-v1/storage/js/copy.mdx [hunk 3]
```diff
@@ -34,19 +31,19 @@ You can copy a file from the specified key to another key within the same [File
...
 // There should now be a new file with key "copied/destKey"
-console.log(list({ path : 'copied/' })); // [ { ..., key: 'copied/destKey' } ]
+console.log(list({ prefix : 'copied/' })); // [ { ..., key: 'copied/destKey' } ]
 console.log(copied); // { key: 'copied/destKey' }
 ```
 
 ## Copy files across access levels
 
-To copy file across different access levels, you'll need to explicitly provide the source and destination access levels:
+To copy a file across different access levels, you'll need to explicitly provide the access level if it's 'private' or 'protected' (Defaults to 'guest').
```
Comment (kvramyasri7) [inline src/fragments/lib-v1/storage/js/copy.mdx:46]: can you please mention that we have to provide accessLevel on both the source and destination if they need to copy across different levels

File: src/fragments/lib-v1/storage/js/getting-started.mdx [hunk 4]
```diff
@@ -35,7 +35,7 @@ When your backend is successfully updated, your new configuration file `aws-expo
 Add Amplify to your app with `yarn` or `npm`:
 
 ```bash
-npm install aws-amplify
+npm install aws-amplify@next
```
Comment (kvramyasri7) [inline src/fragments/lib-v1/storage/js/getting-started.mdx:38]: Nice

File: src/fragments/lib-v1/storage/js/remove.mdx [hunk 5]
```diff
@@ -1,4 +1,5 @@
-Delete stored data from the storage bucket.
+Delete stored data from the storage bucket. To remove a file, you need to provide the key. `accessLevel` can be 'guest', 'private', or 'protected', defaulting to 'guest'.
+
```
Comment (kvramyasri7) [inline src/fragments/lib-v1/storage/js/remove.mdx:2]: We can also add a call out saying that we can only remove from current user and can not `remove` to other users

File: src/fragments/lib-v1/storage/js/copy.mdx (hunk 2 above)
Comment (kvramyasri7) [reply src/fragments/lib-v1/storage/js/copy.mdx:34]: yeah Sure.  There are some merge conflicts can surly do that next time

**Review by kvramyasri7 — APPROVED** [review]

File: src/fragments/lib-v1/storage/js/getting-started.mdx [hunk 6]
```diff
@@ -35,7 +35,7 @@ When your backend is successfully updated, your new configuration file `aws-expo
 Add Amplify to your app with `yarn` or `npm`:
 
 ```bash
```
Comment (kvramyasri7) [inline src/fragments/lib-v1/storage/js/getting-started.mdx:37]: would be nice to say dev-preview is available in `next` tag as well

File: src/fragments/lib-v1/storage/js/copy.mdx (hunk 3 above)
Comment (ashwinkumar6) [reply src/fragments/lib-v1/storage/js/copy.mdx:46]: When coping across access levels I think they need to explicitly provide `accessLevels` only if it's private or protected. I think we would always default to guest in source/destination if `accessLevels` isn't provided

guest (accessLevels isn't required) -> private/protected (accessLevels required)
private/protected (accessLevels required) -> guest (accessLevels isn't required)

File: src/fragments/lib-v1/storage/js/remove.mdx (hunk 5 above)
Comment (ashwinkumar6) [reply src/fragments/lib-v1/storage/js/remove.mdx:2]: I think that's a good point. Thanks will update

File: src/fragments/lib-v1/storage/js/copy.mdx (hunk 3 above)
Comment (kvramyasri7) [reply src/fragments/lib-v1/storage/js/copy.mdx:46]: Yes. But can you please mention to provide accessLevel on both `source` and `destination` if not guest then.

Comment (ashwinkumar6) [reply src/fragments/lib-v1/storage/js/copy.mdx:46]: Ahh i see the suggestion was to add the key words `source` and `destination`. Sorry i thought you were suggesting to mention accessLevels for guest as well. 

This makes sense updated to 
`To copy a file across different access levels, you'll need to explicitly provide the source and destination access levels if it's 'private' or 'protected' (Defaults to 'guest').`

**Review by AllanZhengYP — APPROVED** [review]

**Review by kvramyasri7 — APPROVED** [review]

**Pull request merged**